protocol/message_persistence.go

```diff
@@ -1891,6 +1891,74 @@ func (db sqlitePersistence) MarkMessagesSeen(chatID string, ids []string) (uint6
 	return countWithMentions + countNoMentions, countWithMentions, err
 }
 
+func (db sqlitePersistence) MarkMessageAsUnread(chatID string, messageID string) (uint64, uint64, error) {
+	tx, err := db.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return 0, 0, err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	_, err = tx.Exec(`UPDATE user_messages SET seen = 1 WHERE NOT(seen)`)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	_, err = tx.Exec(
+		`UPDATE user_messages 
+			SET seen = 0 
+			WHERE seen = 1 
+			AND (mentioned OR replied) 
```

Comment:
It took me awhile to get why we do 2 separate requests for `mentioned or replied` messages.
Maybe instead we could do it in a more straightforward way?
1. `UPDATE ... SET seen = 1 WHERE timestamp > ...`
2. `SELECT COUNT(1) FROM user_messages WHERE seen = 0`
3. `SELECT COUNT(1) FROM user_messages WHERE seen = 1`

IMO the intentions would be more obvious like thie.
Also, if I get it right, we will cut the number of db requests from 4 to 3 (if `SELECT changes()` are counted).

Reply:
Ah, I was following the same pattern as other code I learnt from, but let me test having a single query here 👍🏾

Reply:
I tried as my best on this one, but I couldn't make the provided idea work. Let's discuss offline if needed, for now I have added a TODO note for this item, and submitted changes for the other comments to make progress.